### PR TITLE
feat: [CDS-77887]: added base deployment support for ASG

### PR DESCRIPTION
**Pull request by vitsafronovici**
## Describe your changes
In order to use Base Asg configs from CG there are changes on service, infra, step yaml structure that would fit better for both CG and already existing NG. The changes are described in this tech spec https://harness.atlassian.net/wiki/spaces/CDNG/pages/21465530743/AWS+ASG+migration

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52867)

### CODE REVIEW

**Comment by vitsafronovici** [comment]
trigger uts

**Comment by vitsafronovici** [comment]
trigger ut0, ut2, ut4, ut7, ut9

File: 960-api-services/src/main/java/io/harness/aws/asg/manifest/AsgConfigurationManifestHandler.java [hunk 1]
```diff
@@ -234,4 +234,19 @@ AsgInstanceCapacity getRunningInstanceCapacity(
...
+    req.setTags(prepareTags(req.getTags()));
+  }
+
+  List<Tag> prepareTags(Collection<Tag> tags) {
+    if (tags == null) {
+      return null;
+    }
+
+    // keep only key, value properties. For Base deploy resourceId also is provided that leads to error
+    return tags.stream().map(t -> new Tag().withKey(t.getKey()).withValue(t.getValue())).collect(Collectors.toList());
```
Comment (lovish1999) [inline 960-api-services/src/main/java/io/harness/aws/asg/manifest/AsgConfigurationManifestHandler.java:250]: We can even just update this method to always override the resource id with the given asg name ```        if (tag.getResourceId() == null) {
          tag.setResourceId(asgName);
        }. in AsgSdkManager

**Review by vitsafronovici — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52867)* status: 56 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`960-api-services/src/main/java/io/harness/aws/asg/manifest/AsgConfigurationManifestHandler.java` line 250 at r2](https://reviewable.io/reviews/harness/harness-core/52867#gh-1325650639:-NeI-xIAEOKpdVVlff5O:b-s4l1uu) ([raw file](https://github.com/harness/harness-core/blob/64aba423b0bd0127e5dcd5c27ce70a7bf22615d9/960-api-services/src/main/java/io/harness/aws/asg/manifest/AsgConfigurationManifestHandler.java#L250)):*
<details><summary><i>Previously, lovish1999 (Lovish Bansal) wrote…</i></summary><blockquote>

We can even just update this method to always override the resource id with the given asg name ```        if (tag.getResourceId() == null) {
          tag.setResourceId(asgName);
        }. in AsgSdkManager
</blockquote></details>

it does not work like that when using tags from one ASG and apply them to another ASG, even if you update with the needed ASG name as resourceId throwing resource not found

**Review by lovish1999 — DISMISSED** [review]
The changes LGTM.

**Comment by vitsafronovici** [comment]
trigger uts

**Review by acasian-harness — APPROVED** [review]

**Review by lovish1999 — APPROVED** [review]

**Comment by acasian-harness** [comment]
trigger review

**Pull request merged**